src/Tonic/Generator.h

```diff
@@ -76,6 +83,17 @@ namespace Tonic {
       obj->tick(frames, context);
     }
 
+
+  //! Shortcut for wrapping with a Panner. Currently a MonoToStereoPanner but this could/should
+  // be replaced with a better panner that can handle mono or stereo.
+  Generator pan(ControlGenerator pan);
+
+  //! Shortcut for wrapping with a Panner. Currently a MonoToStereoPanner but this could/should
+  // be replaced with a better panner that can handle mono or stereo.
+  Generator pan(float pan);
+
+  void ____recordAdditionAsParameter____(){ obj->hasBeenAdded = true; }
```

Comment:
Hmmm. Not sure I follow you. Do you mean a protected helper method. Or a
helper object? Protected access from a subclass seems problematic if not
impossible:

http://stackoverflow.com/questions/11631777/accessing-a-protected-member-of-a-base-class-in-another-subclass

and friend class inheritance may have the same issues. I'd love to not have
this method public, since it's only ever called inside generator classes.
But it wouldn't surprise me if that just wasn't possible.

On Saturday, March 7, 2015, LB-- <notifications@github.com> wrote:

> In src/Tonic/Generator.h
> <https://github.com/TonicAudio/Tonic/pull/264#discussion_r26001121>:
>
> > @@ -76,6 +83,17 @@ namespace Tonic {
> >        obj->tick(frames, context);
> >      }
> >
> > +
> > +  //! Shortcut for wrapping with a Panner. Currently a MonoToStereoPanner but this could/should
> > +  // be replaced with a better panner that can handle mono or stereo.
> > +  Generator pan(ControlGenerator pan);
> > +
> > +  //! Shortcut for wrapping with a Panner. Currently a MonoToStereoPanner but this could/should
> > +  // be replaced with a better panner that can handle mono or stereo.
> > +  Generator pan(float pan);
> > +
> > +  void ____recordAdditionAsParameter____(){ obj->hasBeenAdded = true; }
>
> You could have a protected helper in the Generator class that accesses
> the member function (basically a proxy). If it's defined inline the
> compiler should optimize it.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/TonicAudio/Tonic/pull/264/files#r26001121>.
>


-- 
===============
Morgan Packard
cell: (720) 891-0122
twitter: @morganpackard